api/data_refinery_api/serializers.py

```diff
@@ -592,35 +592,53 @@ class Meta:
+    there should be at least one experiment.
+    """
+    if data.get("data") is None or type(data["data"]) != dict:
+        raise serializers.ValidationError("`data` must be a dict of lists.")
 
-        if data.get("start") and len(data["data"]) == 0:
-            raise serializers.ValidationError("`data` must contain at least one experiment..")
+    if data.get("start") and len(data["data"]) == 0:
+        raise serializers.ValidationError("`data` must contain at least one experiment..")
 
-        for key, value in data["data"].items():
-            if type(value) != list:
-                raise serializers.ValidationError(
-                    "`data` must be a dict of lists. Problem with `" + str(key) + "`"
-                )
+    accessions = []
+    for key, value in data["data"].items():
+        if type(value) != list:
+            raise serializers.ValidationError(
+                "`data` must be a dict of lists. Problem with `" + str(key) + "`"
+            )
 
-            if len(value) < 1:
-                raise serializers.ValidationError(
-                    "`data` must be a dict of lists, each with one or more elements. Problem with `"
-                    + str(key)
-                    + "`"
-                )
+        if len(value) < 1:
+            raise serializers.ValidationError(
+                "`data` must be a dict of lists, each with one or more elements. Problem with `"
+                + str(key)
+                + "`"
+            )
 
+        try:
+            if len(value) != len(set(value)):
+                raise serializers.ValidationError("Duplicate values detected in " + str(value))
+        except Exception as e:
+            raise serializers.ValidationError("Received bad dataset data: " + str(e))
+
+        # If they want "ALL", just make sure that the experiment has at least one downloadable sample
+        if value == ["ALL"]:
             try:
-                if len(value) != len(set(value)):
-                    raise serializers.ValidationError("Duplicate values detected in " + str(value))
+                experiment = Experiment.processed_public_objects.get(accession_code=key)
             except Exception as e:
-                raise serializers.ValidationError("Received bad dataset data: " + str(e))
+                raise serializers.ValidationError(
+                    "Experiment " + key + " does not have at least one downloadable sample"
+                )
+        # Otherwise, we will check that all the samples they requested are downloadable
+        else:
+            accessions.extend(value)
 
-    else:
-        raise serializers.ValidationError("`data` must be a dict of lists.")
+    if len(accessions) > 0:
+        processed_samples = Sample.processed_objects.filter(accession_code__in=accessions)
+        if len(accessions) != processed_samples.count():
+            raise serializers.ValidationError("Non-downloadable samples in " + str(accessions))
```

Comment:
I think we can be more helpful than this. If we changed the query to be `Sample.public_objects.filter(accession_code__in=accessions, is_processed=False)` then we could check if the result is empty. If it's not then we know exactly which accessions weren't processed and we can instead say "The following samples are not downloadable: "